src/modules/complianceengine/src/lib/LuaEvaluator.cpp

```diff
@@ -0,0 +1,354 @@
+    }
+}
+
+void LuaEvaluator::RegisterProcedures()
+{
+    lua_getfield(L, LUA_REGISTRYINDEX, "restricted_env");
+    if (!lua_istable(L, -1))
+    {
+        lua_pop(L, 1);
+        lua_newtable(L);
+        lua_pushvalue(L, -1);
+        lua_setfield(L, LUA_REGISTRYINDEX, "restricted_env");
+    }
+
+    for (const auto& procedureEntry : Evaluator::mProcedureMap)
+    {
+        const std::string& procedureName = procedureEntry.first;
+        const ProcedureActions& actions = procedureEntry.second;
+
+        if (actions.audit)
+        {
+            std::string auditFunctionName = "Audit" + procedureName;
+            lua_pushstring(L, auditFunctionName.c_str());
+            lua_pushlightuserdata(L, reinterpret_cast<void*>(actions.audit));
+            lua_pushcclosure(L, LuaEvaluator::LuaProcedureWrapper, 2);
+            lua_setfield(L, -2, auditFunctionName.c_str());
+        }
+
+        if (actions.remediate)
+        {
+            std::string remediateFunctionName = "Remediate" + procedureName;
+            lua_pushstring(L, procedureName.c_str());
+            lua_pushlightuserdata(L, reinterpret_cast<void*>(actions.remediate));
+            lua_pushcclosure(L, LuaEvaluator::LuaProcedureWrapper, 2);
+            lua_setfield(L, -2, remediateFunctionName.c_str());
+        }
+    }
+
+    lua_pop(L, 1);
+}
+
+void LuaEvaluator::SecureLuaEnvironment()
+{
+    lua_newtable(L);
+
+    const std::vector<const char*> safeGlobals = {
+        "print", "type", "tostring", "tonumber", "pairs", "ipairs", "next", "pcall", "xpcall", "select", "math"};
+    const std::map<const char*, std::vector<const char*>> safeModuleFunctions = {
+        {"string", {"byte", "char", "find", "format", "gsub", "len", "lower", "match", "rep", "reverse", "sub", "upper"}},
+        {"table", {"concat", "insert", "remove", "sort"}}, {"io", {"lines"}}, {"os", {"time", "date", "clock", "difftime"}}};
+
+    for (const auto& global : safeGlobals)
+    {
+        lua_getglobal(L, global);
+        lua_setfield(L, -2, global);
+    }
+
+    for (const auto& moduleFunctions : safeModuleFunctions)
+    {
+        lua_getglobal(L, moduleFunctions.first);
+        if (lua_istable(L, -1))
+        {
+            lua_newtable(L);
+            for (const auto& func : moduleFunctions.second)
+            {
+                lua_getfield(L, -2, func);
+                lua_setfield(L, -2, func);
+            }
+
+            lua_setfield(L, -3, moduleFunctions.first);
+        }
+        lua_pop(L, 1);
+    }
+
+    lua_pushvalue(L, -1);
+    lua_setfield(L, LUA_REGISTRYINDEX, "restricted_env");
+
+    lua_pop(L, 1);
+}
+
+int LuaEvaluator::LuaProcedureWrapper(lua_State* L)
+{
+    lua_pushstring(L, "lua_call_context");
+    lua_gettable(L, LUA_REGISTRYINDEX);
+    LuaCallContext* callContext = static_cast<LuaCallContext*>(lua_touserdata(L, -1));
+    lua_pop(L, 1);
+
+    if (!callContext)
+    {
+        lua_pushstring(L, "Failed to get call context");
```

Comment:
No log error?

Reply:
At this point we don't have the callContext -> we don't have the log handle